test/it/uk/gov/hmrc/individualsifapistub/IncomeSaRepositorySpec.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it.uk.gov.hmrc.individualsifapistub
+
+import org.scalatest.BeforeAndAfterEach
+import play.api.Configuration
+import reactivemongo.api.indexes.IndexType.Ascending
+import uk.gov.hmrc.individualsifapistub.domain.{DuplicateException, IncomeSaResponse}
+import uk.gov.hmrc.individualsifapistub.repository.IncomeSaRepository
+import uk.gov.hmrc.mongo.MongoSpecSupport
+import unit.uk.gov.hmrc.individualsifapistub.util.TestSupport
+import unit.uk.gov.hmrc.individualsifapistub.util.testUtils.IncomeSaHelpers
+
+class IncomeSaRepositorySpec
+    extends TestSupport
+    with MongoSpecSupport
+    with BeforeAndAfterEach
+    with IncomeSaHelpers {
+
+  override lazy val fakeApplication = buildFakeApplication(
+    Configuration("mongodb.uri" -> mongoUri))
+
+  val repository = fakeApplication.injector.instanceOf[IncomeSaRepository]
+
+  val nino = "XH123456A"
+  val trn = "12345678"
+
+  val innerValue = Seq(createValidSaTaxYearEntry(), createValidSaTaxYearEntry())
+  val request = IncomeSaResponse(Some(innerValue))
+
+  override def beforeEach() {
+    await(repository.drop)
+    await(repository.ensureIndexes)
+  }
+
+  override def afterEach() {
+    await(repository.drop)
+  }
+
+  "collection" should {
+    "have a unique index on nino" in {
+      await(repository.collection.indexesManager.list()).find({ i =>
+        i.name.contains("nino") &&
+          i.key == Seq("id.nino" -> Ascending) &&
+          i.background &&
+          i.unique
+      }) should not be None
```

Comment:
Should we assert what this is rather than it is not 'none'?

Reply:
The returned type is pretty complex and the only this we really care about is that the index exists and as such its simpler and easier to understand by saying that it shouldnt be none e.g. it should exist.